src/Faithlife.Tracing/ListTraceContext.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Faithlife.Tracing
+{
+	public sealed class ListTraceContext : ITraceContext
+	{
+		public ListTraceContext()
+		{
+			m_traces = new List<ITrace>();
+		}
+
+		public ITraceFactory TraceFactory { get; set; }
+
+		public ITrace CurrentTrace => m_traces.LastOrDefault();
+
+		public void Add(ITrace trace)
+		{
+			m_traces.Add(trace);
+		}
+
+		public void Remove(ITrace trace)
+		{
+			if (m_traces.Count > 0)
+			{
+				if (m_traces[m_traces.Count - 1] == trace)
+				{
+					m_traces[m_traces.Count - 1] = null;
+					for (var index = m_traces.Count - 1; index >= 0 && m_traces[index] == null; index--)
```

Comment:
I think I was going to set an entry to `null` (instead of removing it) on line 34, which would have left "holes" of `null`s in the list. So this code is unnecessary right now.